src/domain/mongoose_domain_sql.erl

```diff
@@ -138,34 +143,51 @@ select_from(FromId, Limit) ->
 select_updates_from(FromId, Limit) ->
     Pool = get_db_pool(),
     Args = rdbms_queries:add_limit_arg(Limit, [FromId]),
-    {selected, Rows} = execute_successfully(Pool, domain_select_events_from, Args),
-    Rows.
-
-get_min_event_id() ->
-    Pool = get_db_pool(),
-    selected_to_int(execute_successfully(Pool, domain_events_min, [])).
+    case execute_successfully(Pool, domain_select_events_from, Args) of
+        {selected, []} ->
+            %% get MaxID, this inserts the dummy record
+            %% in the events table if required.
+            MaxId = get_max_event_id(),
+            if
+                MaxId > FromId ->
+                    select_updates_from(FromId, Limit);
+                true ->
+                    %% This must never happen though, unless the events table
+                    %% is modified externally. this is critical error!
+                    %% TODO: figure out what should we do in this case
+                    ?LOG_ERROR(#{what => select_updates_from_failed, limit => Limit,
+                                 from_id => FromId, max_id => MaxId}),
+                    []
+            end;
+        {selected, Rows} -> Rows
+    end.
 
 get_max_event_id() ->
```

Comment:
get_max_event_id_or_set_dummy.

`get_` something means that it does not change anything.